src/openshiftCommonUI.module.js

```diff
@@ -22,6 +22,14 @@ angular.module('openshiftCommonUI', [])
   maxlength: 253,
   description: 'Name must consist of lower-case letters, numbers, periods, and hyphens. It must start and end with a letter or a number.'
 })
+// URI_PATH_FRAGMENT_VALIDATION patterns must consist of following characters:
+// a-z A-Z 0-9 . - _ ~ ! $ & ' ( ) * + , ; = : @
+// https://stackoverflow.com/questions/4669692/valid-characters-for-directory-part-of-a-url-for-short-links
+.constant('URI_PATH_FRAGMENT_VALIDATION', {
+  pattern: /^[a-zA-Z0-9.\-_~!$&'()*+,;=:@]+$/,
+  minlength: 8,
```

Comment:
If this is a generic URI path validator and not specific to webhook secrets, it shouldn't have a minlength here. You would add that check specifically for webhook validation.